src/main/java/org/stocker/PingVerticle.java

```diff
@@ -98,16 +99,16 @@ public void start() {
      * @param updateRecordsRoute
      */
     private void populateDB(UpdateRecordsRoute updateRecordsRoute) {
-        LocalDate localDate = LocalDate.now();
-        for(int i=0 ; i < 10 ; i++){
-            Date date = localDate.minusDays(i).toDate();
-            try {
-                updateRecordsRoute.updateStocksForDate(date);
-            } catch (StockDataNotFoundForGivenDateException e) {
-                logger.warning(" failed for date - " + date.toString());
-            } catch (NseDataObjParseException e) {
-                logger.warning(" failed with parsing error for date - " + date.toString());
-            }
+        Date date = LocalDate.now().toDate();
+
+        try {
+            updateRecordsRoute.updateStocksForDate(date);
+        } catch (StockDataNotFoundForGivenDateException e) {
```

Comment:
just collapse all three into a single catch and log e.getMessage() along with date .